pennylane/devices/default_gaussian.py

```diff
@@ -728,6 +730,12 @@ def apply(self, operation, wires, par):
             self._state = self._operation_map[operation](*par, hbar=self.hbar)
             return  # we are done here
 
+        if operation == "Snapshot":
+            if self.debugger and self.debugger.active:
```

Comment:
`self.debugger` could evaluate to `True` even if it's not a debugger object; in such a case `self.debugger.active` might break. This is a recurring suggestion applicable to other places in the PR too.
```suggestion
            if self.debugger is not None and self.debugger.active:
```

Reply:
I see your point, although checking against not None really is not a guarantee either that `self.debugger` is in any way an object with a `active` attribute. Our code either sets the attribute to None or to a valid `_Debugger` object, so I don't think the change offers any additional protection, but I'm fine changing it as suggested.